### ISSUE TITLE
[Python] Implement new Concept Exercise: tuples

### ISSUE THREAD

**Issue by BethanyG** [exercism/v3#1097]
This issue describes how to implement the `tuples` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the basics of the Tuple ([sequence type](https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range), [immutable Sequence type](https://docs.python.org/3/library/stdtypes.html#immutable-sequence-types)) data type in Python.

## Learning objectives

*  Create a tuple via constructor (`tuple(_iterable_)`), literal (`a,b,c` or `(a, b, c)`), & single literal (`a,` or `(a,)`)
*  Understand that it is the _comma_  in a sequence that makes a Tuple, and that the `()` are optional, except for denoting an _empty_ Tuple or when omitting them creates ambiguity.
*  Understand that Tuples are an _immutable_ data type (like Strings).  Changing a Tuple means creating a new copy.
*  Understand that Tuples can contain other Tuples. (e.g. Tuples can be nested).
*  Create a new Tuple from two or more previous Tuples via concatenation using the `+` operator. 
~*  Create a new Tuple by _multiplying_ a previous Tuple using the `*` operator.~
*  Access values in a tuple via index & using `[]` (_bracket notation_).
*  Check for membership of an item  in a given tuple.
*  Iterate through a tuple using `for item in`.

## Out of scope

*  Sequence type methods such as `min()`, `max()`, `x.index()`, `x.count()`, `len()`, or `hash()`
*  Additional builtin functions as they relate to this data structure (e.g. `sorted()`, `enumerate()`, or  `reversed()`.
*  Slicing or slice notation (`[start:stop:step]`)
*  Knowing that Tuples can be used as objects in other data structures, -- e.g. " a "List of Tuples", "Tuples as keys in Dictionaries", ,or "A Set of Tuples".
*  Create a new Tuple by _multiplying_ a previous Tuple using the `*` operator.
*  Hash-ability and when a Tuple is _not_ hash-able. 
*  Related collections module with `namedtuple()`
*  Memory and performance characteristics.

## Concepts

*  tuples
*  immutability
*  indexing
*  concatenation
*  iteration

## Prerequisites

- [builtin-types](https://github.com/exercism/v3/tree/master/languages/python/reference/concepts/builtin_types)
- [data-types](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/data_structures.md)

## Resources to refer to

*  [Tuples (Python official docs)](https://docs.python.org/3/library/stdtypes.html#tuples)
*  [Sequence Types (Python official docs)](https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range)

### Hints

Hints should link to the Tuples section of the Python docs tutorial: [Tuples](https://docs.python.org/3/tutorial/datastructures.html#tuples-and-sequences).

### After

After, the student can explore slicing syntax, although it will be taught in separate exercises.  This might also be a good time to explore the concept of hash-ability & the use of Tuples as keys in a dictionary.  Additionally, the student can take a look at additional sequence type methods, or how built-in methods such as `sorted()`, `enumerate()`, `min()`, or `max()` can be used with Tuples.

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase, and the test file named dicts_basic_test.py.
Please see [Implementing a Concept Exercise](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md) for additional instructions and details.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
- edited for updated naming by @yawpitch
- edited problem description to include nested tuples but _exclude_ `len()` as out of scope.
- removed `*` multiplication from learning objectives.
